auth-jwt/Sources/App/Middleware/JWTAuthenticator.swift

```diff
@@ -0,0 +1,48 @@
+import Foundation
+import Hummingbird
+import HummingbirdAuth
+import JWTKit
+
+struct JWTPayloadData: JWTPayload, Equatable, HBAuthenticatable {
+  enum CodingKeys: String, CodingKey {
+    case subject = "sub"
+    case expiration = "exp"
+  }
+
+  var subject: SubjectClaim
+  var expiration: ExpirationClaim
+  // Define additional JWT Attributes here
+
+  func verify(using signer: JWTSigner) throws {
+    try self.expiration.verifyNotExpired()
+  }
+}
+
+struct JWTAuthenticator: HBAsyncAuthenticator {
+  var jwks: JWKS
+
+  init(jwksUrl: String) {
+    do {
+      let jwksKeys = URL(string: jwksUrl)!
+      let jwksData = try Data(contentsOf: jwksKeys)
+      jwks = try JSONDecoder().decode(JWKS.self, from: jwksData)
+    } catch {
+      print("middleware initialization failed")
```

Comment:
The problem you are getting here is because you are leaving the `init` function without actually giving `jwks` a valid value. I think you would be better off having the `init` throw the error instead of trying to catch it. 
```swift
init(jwksUrl: String) throws {
    let jwksKeys = URL(string: jwksUrl)!
    let jwksData = try Data(contentsOf: jwksKeys)
    jwks = try JSONDecoder().decode(JWKS.self, from: jwksData)
}
```
Then where you try to construct the `JWTAuthenticator` you catch the error and print the error
```swift
let jwtAuthenticator: JWTAuthenticator
do {
    jwtAuthenticator = try JWTAuthenticator(jwksUrl: jwksUrl)
} catch {
    print("JWTAuthenticator initialization failed")
    throw error
}